airflow/models/trigger.py

```diff
@@ -184,7 +184,10 @@ def assign_unassigned(cls, triggerer_id, capacity, session=None):
         # Find triggers who do NOT have an alive triggerer_id, and then assign
         # up to `capacity` of those to us.
         trigger_ids_query = (
-            session.query(cls.id).filter(cls.triggerer_id.notin_(alive_triggerer_ids)).limit(capacity).all()
+            session.query(cls.id)
+            .filter(or_(cls.triggerer_id.notin_(alive_triggerer_ids), cls.triggerer_id == None))  # noqa: E711
```

Comment:
no, unfortunately that's a quirk with sqlalchemy's `notin_` (test fails when that's removed). I'll add a comment to that effect

Reply:
ok makes sense.

small thing but  can i also suggest in that  case, put the cls.triggerer_id.is_(None)  _first_ in the OR

then it reads more like "not assigned OR on a dead triggerer"
compared with "on a dead triggerer OR not assigned"

"not assigned" is by far the more common / standard / expected case i think so seems like it would be more intuitive to put it first.

Reply:
make sense to me